crates/loans/src/lib.rs

```diff
@@ -1731,9 +1730,9 @@ impl<T: Config> Pallet<T> {
     // Returns `Err` If InsufficientLiquidity
     // `account`: account that need a liquidity check
     // `reduce_amount`: values that will have an impact on liquidity
-    fn ensure_liquidity(account: &T::AccountId, reduce_amount: FixedU128) -> DispatchResult {
-        let (total_liquidity, _) = Self::get_account_liquidity(account)?;
-        if total_liquidity >= reduce_amount {
+    fn ensure_liquidity(account: &T::AccountId, reduce_amount: Amount<T>) -> DispatchResult {
+        let AccountLiquidity { liquidity, .. } = Self::get_account_liquidity(account)?;
```

Comment:
Let me know if you think it's cleaner to add getter functions for `liquidity` and `shortfall` to the impl of `AccountLiquidity`.